src/service/resources-impl.js

```diff
@@ -794,7 +794,10 @@ export class Resources {
             minTop = minTop == -1 ? box.top : Math.min(minTop, box.top);
           }
           request.resource./*OK*/changeSize(
-              request.newHeight, request.newWidth, request.callback);
+              request.newHeight, request.newWidth);
+          if (request.callback) {
+            request.callback(/* hasSizeChanged */true);
```

Comment:
Instead of calling in several places, we can put this at the end of this (big) conditional.